shell/platform/windows/win32_window.cc

```diff
@@ -190,35 +191,71 @@ Win32Window::MessageHandler(HWND hwnd,
         // DefWindowProc will send WM_CHAR for this WM_UNICHAR.
         break;
       }
+      case WM_DEADCHAR:
+      case WM_SYSDEADCHAR:
       case WM_CHAR:
       case WM_SYSCHAR: {
-        if (wparam == VK_BACK)
-          break;
         char32_t code_point = static_cast<char32_t>(wparam);
         static char32_t lead_surrogate = 0;
-        // If code_point is LeadSurrogate, save and return.
+        // If code_point is LeadSurrogate, save to combine to potentially form
+        // a complex Unicode character.
         if ((code_point & 0xFFFFFC00) == 0xD800) {
           lead_surrogate = code_point;
-          return TRUE;
         }
         // Merge TrailSurrogate and LeadSurrogate.
         if (lead_surrogate != 0 && (code_point & 0xFFFFFC00) == 0xDC00) {
```

Comment:
For clarity, I would make this an else-if, since it's odd to have the first condition pass due to the code that just established that it was a lead surrogate. (The second part of the check will fail anyway, but I had to read it closely to be sure we weren't accidentally running both blocks.)